stacker/context.py

```diff
@@ -30,62 +32,80 @@ class Context(object):
     the command line and specified in the config to `Stack` objects.
 
     Args:
-        namespace (str): A unique namespace for the stacks being built.
         environment (dict): A dictionary used to pass in information about
             the environment. Useful for templating.
         stack_names (list): A list of stack_names to operate on. If not passed,
             usually all stacks defined in the config will be operated on.
-        mappings (dict): Used as Cloudformation mappings for the blueprint.
         config (dict): The configuration being operated on, containing the
             stack definitions.
         force_stacks (list): A list of stacks to force work on. Used to work
             on locked stacks.
 
     """
 
-    def __init__(self, environment,  # pylint: disable-msg=too-many-arguments
+    def __init__(self, environment=None,
                  stack_names=None,
-                 mappings=None,
                  config=None,
                  logger_type=None,
                  force_stacks=None):
-        try:
-            self.namespace = environment["namespace"]
-        except KeyError:
-            raise MissingEnvironment(["namespace"])
-
         self.environment = environment
         self.stack_names = stack_names or []
-        self.mappings = mappings or {}
         self.logger_type = logger_type
-        self.namespace_delimiter = "-"
         self.config = config or {}
         self.force_stacks = force_stacks or []
-        self._base_fqn = self.namespace.replace(".", "-").lower()
-        self.bucket_name = "stacker-%s" % (self.get_fqn(),)
-        self.tags = {
-            'stacker_namespace': self.namespace
-        }
-
         self.hook_data = {}
 
+    @property
+    def namespace(self):
+        namespace = self.config.get("namespace")
+        if namespace is not None:
+            return namespace
+
+        # For backwards compatibility, we fallback to attempting to
+        # fetch the namespace from the environment file, if provided.
+        namespace = self.environment.get("namespace")
+        if namespace:
+            logger.warn("specifying namespace in the environment is "
+                        "deprecated, and should be moved to the config")
+            return namespace
+
+        # No namespace
+        return ""
```

Comment:
This feels like it will have far ranging consequences - have you looked at all the places where this might cause issues?  Why would we want to allow a blank namespace?  Seems like an easy way for folks to shoot themselves in the foot.